applications/CompressiblePotentialFlowApplication/custom_elements/adjoint_finite_difference_potential_flow_element.h

```diff
@@ -0,0 +1,118 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//
+//  Main authors:    Marc Nuñez, based on A. Geiser, M. Fusseder, I. Lopez and R. Rossi work
+//
+
+#if !defined(KRATOS_ADJOINT_FINITE_DIFFERENCE_POTENTIAL_FLOW_ELEMENT_H_INCLUDED )
+#define KRATOS_ADJOINT_FINITE_DIFFERENCE_POTENTIAL_FLOW_ELEMENT_H_INCLUDED
+
+
+// Project includes
+#include "adjoint_base_potential_flow_element.h"
+
+namespace Kratos
+{
+
+template <class TPrimalElement>
+class AdjointFiniteDifferencePotentialFlowElement : public AdjointBasePotentialFlowElement<TPrimalElement>
+{
+public:
+
+    ///@name Type Definitions
+    ///@{
+
+    typedef AdjointBasePotentialFlowElement<TPrimalElement> BaseType;
+    static constexpr int NumNodes = TPrimalElement::TNumNodes;
+    static constexpr int Dim = TPrimalElement::TDim;
+    typedef typename BaseType::GeometryType GeometryType;
+    typedef typename BaseType::PropertiesType PropertiesType;
+    typedef typename BaseType::NodesArrayType NodesArrayType;
+
+    ///@}
+    ///@name Pointer Definitions
+    /// Pointer definition of AdjointFiniteDifferencePotentialFlowElement
+    KRATOS_CLASS_INTRUSIVE_POINTER_DEFINITION(AdjointFiniteDifferencePotentialFlowElement);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /**
+     * Constructor.
+     */
+    AdjointFiniteDifferencePotentialFlowElement(IndexType NewId = 0)
```

Comment:
[https://stackoverflow.com/a/121163](https://stackoverflow.com/a/121163)

```suggestion
    explicit AdjointFiniteDifferencePotentialFlowElement(IndexType NewId = 0): AdjointBasePotentialFlowElement<TPrimalElement>(NewId){};
```